app/scripts/components/common/catalog/controls/hooks/use-catalog-view.ts

```diff
@@ -17,6 +25,44 @@ export function useCatalogView() {
   return {
     search,
     taxonomies,
-    onAction
+    onAction,
   };
-}
+} 
+// @NOTE: A hook using qs-state for query parameter management for cross-page navigation
+// Details: https://github.com/NASA-IMPACT/veda-ui/pull/1021
+export function useCatalogViewQS(): UseCatalogViewResult {
+  const navigate = useNavigate();
+  const useQsState = useQsStateCreator({
+    commit: navigate
+  });
+
+  const [search, setSearch] = useQsState.memo(
```

Comment:
Oh, That sounds like how the current main branch is using the filters with the states in the data selection modal, but in a separate hook. Am I getting it right? 

Aside from how we will handle the query parameter in the future... I wonder if there is a better place than the code base where this kind of discussion should go so it can get deserving attention. I changed `@NOTE` prefix to `@TECH-DEBT`to signal that actions need to be taken.

Reply:
Not sure how its managed in the modal but yep talking about not combining taxonomies state and url changes likes whats happening in the `atomWithUrlStability` and to have them separate